raster/src/main/scala/geotrellis/raster/interpolation/KrigingInterpolation.scala

```diff
@@ -0,0 +1,62 @@
+package geotrellis.raster.interpolation
+
+import geotrellis.raster._
+import geotrellis.vector._
+import spire.syntax.cfor._
+
+abstract sealed class KrigingInterpolationMethod
+case object KrigingSimple extends KrigingInterpolationMethod
+case object KrigingOrdinary extends KrigingInterpolationMethod
+case object KrigingUniversal extends KrigingInterpolationMethod
+case object KrigingGeo extends KrigingInterpolationMethod
+
+
+object KrigingInterpolationMethod {
+  val DEFAULT = KrigingOrdinary
+}
+
+abstract class KrigingInterpolation(method: KrigingInterpolationMethod, points: Seq[PointFeature[Int]], re: RasterExtent,
```

Comment:
I see that you're basing your design on classes a lot, which is a very Object Oriented way of doing this. I recommend you try to think about, when does it make sense to just write something as a function, and when does something really benefit from being a class. I'm not asking for a change, but just want you to keep it in mind as you code. Scala in general and GeoTrellis in particular prefers a functional approach, which often eschews stateful instances for a simple idempotent side effect free transformation (e.g. an `object` method). If the design definitely benefits from having classes, or from having inheritance, then we definitely do it when it makes sense. But preferring functions over classes, and composition over inheritance, is a good practice.